qiskit/qpy/__init__.py

```diff
@@ -105,9 +105,203 @@
 Version 5
 =========
 
-Version 5 changes from :ref:`qpy_version_4` by changing two payloads the INSTRUCTION metadata
-payload and the CUSTOM_INSTRUCTION block. These now have new fields to better account for
-:class:`~.ControlledGate` objects in a circuit.
+Version 5 changes from :ref:`qpy_version_4` by adding support for :class:`.~ScheduleBlock`
+and changing two payloads the INSTRUCTION metadata payload and the CUSTOM_INSTRUCTION block.
+These now have new fields to better account for :class:`~.ControlledGate` objects in a circuit.
+In addition, wew payload MAP_ITEM is defined to implement the :ref:`qpy_mapping` block.
+
+In QPY version 5 and above,
+
+.. code-block:: c
+
+    struct {
+        char type;
+    }
+
+immediately follows the file header block to represent the program type stored in the file.
+
+- When ``type==c``, :class:`~.QuantumCircuit` payload follows
+- When ``type==s``, :class:`~.ScheduleBlock` payload follows
+
+Note that different programs cannot be packed together in the same file.
+
+.. _qpy_schedule_block:
+
+SCHEDULE_BLOCK
+--------------
+
+:class:`~.ScheduleBlock` is first supported in QPY Version 5. This allows
+users to save pulse programs in the QPY binary format as follows:
+
+.. code-block:: python
+
+    from qiskit import pulse, qpy
+
+    with pulse.build() as schedule:
+        pulse.play(pulse.Gaussian(160, 0.1, 40), pulse.DriveChannel(0))
+
+    with open('schedule.qpy', 'wb') as fd:
+        qpy.dump(qc, fd)
+
+    with open('schedule.qpy', 'rb') as fd:
+        new_qc = qpy.load(fd)[0]
```

Comment:
I feel like this should be in the release note too. It's good to show people how to use it there